script.js

```diff
@@ -93,3 +98,72 @@ function removeB() {
   const removeId = parseInt(window.event.target.id.replace('remove', ''), 10);
   myShelf.removeBook(removeId);
 }
+
+function displayPage(link) {
+  const listPage = document.getElementById('all-books');
+  const addNewPage = document.getElementById('add-new-page');
+  const contactPage = document.getElementById('contact-page');
+  if (link === 'list') {
+    listPage.classList.remove('hide');
+    contactPage.classList.remove('d-flex');
+    contactPage.classList.add('hide');
+    addNewPage.classList.remove('d-flex');
+    addNewPage.classList.add('hide');
+  } else if (link === 'add-new') {
+    addNewPage.classList.remove('hide');
+    listPage.classList.remove('d-flex');
+    listPage.classList.add('hide');
+    contactPage.classList.remove('d-flex');
+    contactPage.classList.add('hide');
+  } else if (link === 'contact') {
+    contactPage.classList.remove('hide');
+    contactPage.classList.add('d-flex');
+    listPage.classList.remove('d-flex');
+    listPage.classList.add('hide');
+    addNewPage.classList.remove('d-flex');
+    addNewPage.classList.add('hide');
+  }
+}
+
+displayPage('list');
+
+// Clock on click - date and time
+Number.prototype.pa = function (digit) {
+  for (n = this.toString(); n.length < digit; n = 0 + n);
+  return n;
+};
+
+function updateClock() {
+  const at = new Date();
+  const daname = at.getDay();
+  mont = at.getMonth();
+  denum = at.getDate();
+  yrr = at.getFullYear();
+  hourr = at.getHours();
+  min = at.getMinutes();
+  sec = at.getSeconds();
+  pe = 'AM';
+
+  if (hourr >= 12) {
+    pe = 'PM';
+  }
+  if (hourr === 0) {
+    hourr = 12;
+  }
+  if (hourr > 12) {
+    hourr -= 12;
+  }
+
+  const months = ['January', 'February', 'March', 'April', 'May', 'June', 'July', 'August', 'September', 'October', 'November', 'December'];
+  const week = ['Sunday', 'Monday', 'Tuesday', 'Wednesday', 'Thursday', 'Friday', 'Saturday'];
+  const id = ['days', 'monthly', 'daynumb', 'years', 'hours', 'minutes', 'seconds', 'ampm'];
+  const values = [week[daname], months[mont], denum, yrr, hourr.pa(2), min.pa(2), sec, pe];
+
+  for (let i = 0; i < id.length; i += 1) { document.getElementById(id[i]).firstChild.nodeValue = values[i]; }
+}
+
+function initClock() {
+  updateClock();
+  // eslint-disable-next-line no-implied-eval
+  window.setInterval('updateClock()', 1);
```

Comment:
- [ ] [OPTIONAL] Is there a reason why you have passed the callback to `setInterval` as a string? I tried to pass it like `window.setInterval(updateClock, 1);` and it seems to work fine. You don't need to call the callback you pass to `setInterval`, it knows how to handle it, so please consider the approach I showed.

Reply:
Yes, you are right. I changed it.